### DISCUSSION TITLE
Strange 'ORA-01002: fetch out of sequence' issue

### DISCUSSION THREAD

**Discussion by stevefan1999-personal** [oracle/node-oracledb#1583] [Q&A]
I tried to use the latest oracledb driver for Directus, but there is some problem during migration. It seems like the fetch cursor was invalidated, indicating something somewhere we have a concurrency problem.

Check this out
https://github.com/directus/directus/pull/19067#issuecomment-1628378328

**Comment by sharadraju** [comment]
Thanks @stevefan1999-personal for testing our thin mode. What is the Oracle DB version that you are using?

I can see that Knex framework is used. Are you able to isolate the exact the SQL statement that causes this ORA-01002 error.
An isolated test case will help us immensely!

**Reply by stevefan1999-personal** [reply]
https://github.com/stevefan1999-personal/directus/blob/8c23245b47394a978888e7c502630c691666c0ab/api/src/database/migrations/20210802A-replace-groups.ts#L6-L7
Well but you can quickly tell this may not be the root cause but rather the surface of it. This is where the migration up process ends up with. I'm trying to run the Knex connection with a connection pool maximum size of 1, and see if there is any issue.

**Reply by sharadraju** [reply]
@stevefan1999-personal Were you able to run the Knex connection with a connection pool of maxSize 1?
Also do let us know, if you can isolate this issue and provide a simple, runnable test case. That would be extremely helpful.

**Reply by stevefan1999-personal** [reply]
> @stevefan1999-personal Were you able to run the Knex connection with a connection pool of maxSize 1?
> Also do let us know, if you can isolate this issue and provide a simple, runnable test case. That would be extremely helpful.

Let me check this weekend

**Reply by hinogi** [reply]
https://github.com/typeorm/typeorm/pull/10285 seems to have the same problem with thin

**Reply by stevefan1999-personal** [reply]
Seems like there is indeed a concurrency issue

**Reply by sharadraju** [reply]
Closing this discussion as it has been fixed
